go-controller/pkg/kubevirt/pod.go

```diff
@@ -0,0 +1,59 @@
+package kubevirt
+
+import (
+	"fmt"
+
+	corev1 "k8s.io/api/core/v1"
+	metav1 "k8s.io/apimachinery/pkg/apis/meta/v1"
+
+	kubevirtv1 "kubevirt.io/api/core/v1"
+
+	"github.com/ovn-org/ovn-kubernetes/go-controller/pkg/factory"
+	"github.com/ovn-org/ovn-kubernetes/go-controller/pkg/util"
+)
+
+// PodIsLiveMigratable will return true if the pod belongs
+// to kubevirt and should use the live migration features
+func PodIsLiveMigratable(pod *corev1.Pod) bool {
+	_, ok := pod.Annotations[kubevirtv1.AllowPodBridgeNetworkLiveMigrationAnnotation]
+	return ok
+}
+
+// FindVMRelatedPods will return pods belong to the same vm annotated at pod
+func FindVMRelatedPods(client *factory.WatchFactory, pod *corev1.Pod) ([]*corev1.Pod, error) {
+	vmName, ok := pod.Labels[kubevirtv1.VirtualMachineNameLabel]
+	if !ok {
+		return []*corev1.Pod{}, nil
+	}
+	vmPods, err := client.GetPodsBySelector(pod.Namespace, metav1.LabelSelector{MatchLabels: map[string]string{kubevirtv1.VirtualMachineNameLabel: vmName}})
+	if err != nil {
+		return []*corev1.Pod{}, err
+	}
+	return vmPods, nil
+}
+
+// FindNetworkInfo will return the original switch name and the OVN pod
+// annotation from any other pod annotated with the same VM as pod
+func FindNetworkInfo(client *factory.WatchFactory, pod *corev1.Pod) (NetworkInfo, error) {
+	vmPods, err := FindVMRelatedPods(client, pod)
+	if err != nil {
+		return NetworkInfo{}, fmt.Errorf("failed finding related pods for pod %s/%s when looking for network info: %v", pod.Namespace, pod.Name, err)
+	}
+	networkInfo := NetworkInfo{
+		OriginalSwitchName: pod.Spec.NodeName,
```

Comment:
The only pending pod would be the one running the VM and we are covering this with 
```golang
if vmPod.Name == pod.Name {                                             
            continue                                                            
        }      
```

So all the rest of the pods are going to be "Running" or "Complete"

Reply:
I'm not seeing the code you pasted, but I see:

```
	if len(vmPods) == 0 || vmPods[0].Name == pod.Name {
		return networkInfo, nil
	}
```

I'm not sure how this covers my comment. If the new pod you are adding is not yet scheduled, then it could have an empty nodeName and set it as the OriginalSwitchName. Perhaps the calling functions protect against his case, so it is not a real problem, but might be good to add a check here.

Also, why do you compare ```vmPods[0].Name == pod.Name``` and then return. Do we expect migrated pods to not have the same name?

Reply:
> I'm not seeing the code you pasted, but I see:
> 
> ```
> 	if len(vmPods) == 0 || vmPods[0].Name == pod.Name {
> 		return networkInfo, nil
> 	}
> ```

Is the original code that loops over the pods, new version just take the first one that is not the same pod.

> 
> I'm not sure how this covers my comment. If the new pod you are adding is not yet scheduled, then it could have an empty nodeName and set it as the OriginalSwitchName. Perhaps the calling functions protect against his case, so it is not a real problem, but might be good to add a check here.

If OriginalSwitchName is empty caller will fail searching for the swich subnet and the pod will be "reconcile" later on, same as original code where NodeName is empty and pod has to be reconcile later on to get the switch subnet.

> 
> Also, why do you compare `vmPods[0].Name == pod.Name` and then return. Do we expect migrated pods to not have the same name?

The `FindVMRelatedPods` just return all the pods realted to the same VM including the pod you pass as hint.